orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -73,13 +74,76 @@ def subscriptions_all() -> List[SubscriptionTable]:
     return SubscriptionTable.query.all()
 
 
+def _update_in(dct: Union[dict, list], path: str, value: Any, sep: str = ".") -> None:
+    """Update a value in a dict or list based on a path."""
+    for x in path.split(sep):
+        prev: Union[dict, list]
+        if x.isdigit() and isinstance(dct, list):
+            prev = dct
+            dct = dct[int(x)]
+        else:
+            prev = dct
+            dct = dict(dct).setdefault(x, {})
+    prev[x] = value  # type: ignore
+
+
+def _get_in(dct: Union[dict, list], path: str, sep: str = ".") -> Any:
+    """Get a value in a dict or list using the path and get the resulting key's value."""
+    prev: Union[dict, list]
+    for x in path.split(sep):
+        if x.isdigit() and isinstance(dct, list):
+            prev, dct = dct, dct[int(x)]
+        else:
+            prev, dct = dct, dict(dct).get(x)  # type: ignore
+    return prev[x]  # type: ignore
+
+
+def _getattr_in(obj: Any, attr: str, *args: List[Any]) -> Any:
+    """Get an instance attribute value by path."""
+
+    def _getattr(obj: object, attr: str) -> Any:
+        return getattr(obj, attr, None, *args)
+
+    return functools.reduce(_getattr, [obj] + attr.split("."))
+
+
+# def _updateattr_in(obj: Any, attr: str, val: Any) -> None:
+#     """Set an instance attribute value by path."""
+#     pre, _, post = attr.rpartition(".")
+#     return setattr(_getattr_in(obj, pre) if pre else obj, post, val)
+
+
+def product_block_paths(subscription: SubscriptionModel) -> List[Optional[str]]:
+    def get_dict_items(d: dict) -> Generator:
+        for k, v in d.items():
+            if isinstance(v, dict):
+                for k1, v1 in get_dict_items(v):
+                    yield (f"{k}.{k1}", v1)
+                yield (k, v)
+
+    return [c[0] for c in get_dict_items(subscription.dict())]
```

Comment:
A unit test for this function would be very useful

Reply:
And these functions might be better placed in a utility module like `api/helpers.py` to keep the endpoints module tidy